packages/interactivity/src/proxies/test/state-proxy.ts

```diff
@@ -0,0 +1,1187 @@
+			expect( state.nested ).toEqual( { b: 2 } );
+			expect( state.array ).toEqual( [ 3, { b: 2 } ] );
+			expect( state.array[ 1 ] ).toEqual( { b: 2 } );
+		} );
+
+		it( 'should return plain primitives', () => {
+			expect( state.a ).toBe( 1 );
+			expect( state.nested.b ).toBe( 2 );
+			expect( state.array[ 0 ] ).toBe( 3 );
+			expect(
+				typeof state.array[ 1 ] === 'object' && state.array[ 1 ].b
+			).toBe( 2 );
+			expect( state.array.length ).toBe( 2 );
+		} );
+
+		it( 'should support reading from getters', () => {
+			const state = proxifyState( 'test', {
+				counter: 1,
+				get double() {
+					return state.counter * 2;
+				},
+			} );
+			expect( state.double ).toBe( 2 );
+			state.counter = 2;
+			expect( state.double ).toBe( 4 );
+		} );
+
+		it( 'should support getters returning other parts of the state', () => {
+			const state = proxifyState( 'test', {
+				switch: 'a',
+				a: { data: 'a' },
+				b: { data: 'b' },
+				get aOrB() {
+					return state.switch === 'a' ? state.a : state.b;
+				},
+			} );
+			expect( state.aOrB.data ).toBe( 'a' );
+			state.switch = 'b';
+			expect( state.aOrB.data ).toBe( 'b' );
+		} );
+
+		it( 'should support getters using ownKeys traps', () => {
+			const state = proxifyState( 'test', {
+				x: {
+					a: 1,
+					b: 2,
+				},
+				get y() {
+					return Object.values( state.x );
+				},
+			} );
+
+			expect( state.y ).toEqual( [ 1, 2 ] );
+		} );
+
+		it( 'should support getters accessing the scope', () => {
+			const state = proxifyState( 'test', {
+				get y() {
+					const ctx = getContext< { value: string } >();
+					return ctx.value;
+				},
+			} );
+
+			const scope = { context: { test: { value: 'from context' } } };
+			try {
+				setScope( scope as any );
+				setNamespace( 'test' );
+				expect( state.y ).toBe( 'from context' );
+			} finally {
+				resetNamespace();
+				resetScope();
+			}
+		} );
+
+		it( 'should work with normal functions', () => {
+			const state = proxifyState( 'test', {
+				value: 1,
+				isBigger: ( newValue: number ): boolean =>
+					state.value < newValue,
+				sum( newValue: number ): number {
+					return state.value + newValue;
+				},
+				replace: ( newValue: number ): void => {
+					state.value = newValue;
+				},
+			} );
+			expect( state.isBigger( 2 ) ).toBe( true );
+			expect( state.sum( 2 ) ).toBe( 3 );
+			expect( state.value ).toBe( 1 );
+			state.replace( 2 );
+			expect( state.value ).toBe( 2 );
+		} );
+
+		it( 'should work with normal functions accessing the scope', () => {
+			const state = proxifyState( 'test', {
+				sumContextValue( newValue: number ): number {
+					const ctx = getContext< { value: number } >();
+					return ctx.value + newValue;
+				},
+			} );
+
+			const scope = { context: { test: { value: 1 } } };
+			try {
+				setScope( scope as any );
+				setNamespace( 'test' );
+				expect( state.sumContextValue( 2 ) ).toBe( 3 );
+			} finally {
+				resetNamespace();
+				resetScope();
+			}
```

Comment:
Same here.

```suggestion
			try {
				setScope( scope as any );
				expect( state.sumContextValue( 2 ) ).toBe( 3 );
			} finally {
				resetScope();
			}
```

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/62734/commits/8a6589aba4a69b7f339a327ffa3543f2b3ea8ef0